nautobot/utilities/fields.py

```diff
@@ -72,3 +82,87 @@ def deconstruct(self):
             [self.target_field],
             kwargs,
         )
+
+
+class JSONArrayField(models.JSONField):
+    """
+    An ArrayField implementation backed JSON storage.
+    Replicates ArrayField's base field validation.
+    """
+
+    _default_hint = ("list", "[]")
+
+    def __init__(self, base_field, **kwargs):
+        self.base_field = base_field
+        super().__init__(**kwargs)
+
+    def set_attributes_from_name(self, name):
+        super().set_attributes_from_name(name)
+        self.base_field.set_attributes_from_name(name)
+
+    @property
+    def description(self):
+        return "JSON Array of %s" % self.base_field.description
+
+    def get_prep_value(self, value):
+        if value is not None:
+            if not isinstance(value, (list, tuple)):
+                raise ValueError("value {} is not list or tuple".format(value))
+            value = [self.base_field.get_prep_value(v) for v in value]
+        return super().get_prep_value(value)
+
+    def deconstruct(self):
+        name, path, args, kwargs = super().deconstruct()
+        kwargs.update(
+            {
+                "base_field": self.base_field.clone(),
+            }
+        )
+        return name, path, args, kwargs
+
+    def to_python(self, value):
+        if isinstance(value, str):
+            # Assume we're deserializing
+            vals = json.loads(value)
+            value = [self.base_field.to_python(val) for val in vals]
+        return value
+
+    def value_to_string(self, obj):
+        values = []
+        vals = self.value_from_object(obj)
+        base_field = self.base_field
+
+        for val in vals:
+            if val is None:
+                values.append(None)
+            else:
+                obj = AttributeSetter(base_field.attname, val)
+                values.append(base_field.value_to_string(obj))
+        return json.dumps(values)
+
+    def validate(self, value, model_instance):
+        super().validate(value, model_instance)
+        if isinstance(self.base_field, JSONArrayField):
+            raise exceptions.ValidationError("cannot nest JSONArrayFields")
+        for index, part in enumerate(value):
```

Comment:
It doesn't look like `index` is used:

```suggestion
        for part in value:
```